src/main/java/ui/components/pickers/LabelPickerState.java

```diff
@@ -0,0 +1,338 @@
+package ui.components.pickers;
+
+import backend.resource.TurboLabel;
+import util.Utility;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * This class is used to represent the state of the label picker. In additional to representing a state,
+ * it contains the logic which handles toggling/highlighting of labels and simplifies the retrieval of
+ * the resulting status for UI to display.
+ */
+public class LabelPickerState {
+    Set<String> initialLabels;
+    List<String> addedLabels;
+    List<String> removedLabels;
+    List<String> matchedLabels;
+    OptionalInt currentSuggestionIndex;
+
+    /*
+     * Constructors
+     */
+
+    public LabelPickerState(Set<String> initialLabels) {
+        this.initialLabels = initialLabels;
+        this.addedLabels = new ArrayList<>();
+        this.removedLabels = new ArrayList<>();
+        this.matchedLabels = new ArrayList<>();
+        this.currentSuggestionIndex = OptionalInt.empty();
+    }
+
+    private LabelPickerState(Set<String> initialLabels,
+                             List<String> addedLabels,
+                             List<String> removedLabels,
+                             List<String> matchedLabels,
+                             OptionalInt currentSuggestionIndex) {
+        this.initialLabels = initialLabels;
+        this.addedLabels = addedLabels;
+        this.removedLabels = removedLabels;
+        this.matchedLabels = matchedLabels;
+        this.currentSuggestionIndex = currentSuggestionIndex;
+    }
+
+    private LabelPickerState(Set<String> initialLabels,
+                             List<String> addedLabels,
+                             List<String> removedLabels,
+                             List<String> matchedLabels) {
+        this.initialLabels = initialLabels;
+        this.addedLabels = addedLabels;
+        this.removedLabels = removedLabels;
+        this.matchedLabels = matchedLabels;
+        this.currentSuggestionIndex = OptionalInt.empty();
+    }
```

Comment:
In cases like this, it is common for constructors to call each other. Those with fewer parameters will call the ones with more parameters.
e.g. 
```java
public LabelPickerState(Set<String> initialLabels) {
        LabelPickerState(initialLabels, new ArrayList<>(), new ArrayList<>(), new ArrayList<>(), OptionalInt.empty());
}
```